src/components/Messaging/AnnounceMessage.js

```diff
@@ -0,0 +1,217 @@
+    style: {
+      maxHeight: 48 * 4.5 + 8,
+      width: 250,
+    },
+  },
+};
+
+const AnnounceMessageForm = () => {
+  const { orgList } = useContext(AppContext);
+  const { user, regions, postMessageSend } = useContext(MessagingContext);
+  const { form, sendButton } = useStyles();
+  const [values, setValues] = useState({
+    message: '',
+    videoLink: '',
+    organization: '',
+    region: '',
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setValues({
+      ...values,
+      [name]: value,
+    });
+  };
+  const { message, videoLink, organization, region } = values;
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    const payload = {
+      author_handle: user.userName,
+      subject: 'Announce Message',
+      body: values.message,
+      // video_link: values.videoLink,
+    };
+    if (region) {
+      payload['region_id'] = region;
+    }
+    if (organization) {
+      payload['organization_id'] = organization;
+    }
+    if (payload.organization_id || payload.region_id) {
+      console.log(payload);
+      await postMessageSend(payload);
+    }
+  };
+
+  return (
+    <form className={form} onSubmit={handleSubmit}>
+      <GSInputLabel text={'Message'} />
+      <TextField
+        multiline
+        rows={5}
+        name="message"
+        value={message}
+        onChange={handleChange}
+        label="Write your message here ..."
+      />
+      <GSInputLabel text={'Add a Video Link'} />
+      <TextField
+        name="videoLink"
+        value={videoLink}
+        onChange={handleChange}
+        label="Add a video link e.g. youtube url"
+      />
+      <GSInputLabel
+        id="select-label"
+        text={'Target Audience by Organization'}
+      />
+      <Select
+        labelId="select-label"
+        id="select"
+        label="Select Organization"
+        name="organization"
+        value={organization}
+        onChange={handleChange}
+        input={<OutlinedInput />}
+        MenuProps={MenuProps}
+      >
+        {orgList.map((org, i) => (
+          <MenuItem key={i} value={org.id}>
+            {org.name}
+          </MenuItem>
+        ))}
+      </Select>
+      <GSInputLabel id="select-reg-label" text={'Target Audience by Region'} />
+      <Select
+        labelId="select-reg-label"
+        label="Regions"
+        name="region"
+        value={region}
+        onChange={handleChange}
+        input={<OutlinedInput />}
+        id="select-reg"
+      >
+        {regions.map((region) => (
+          <MenuItem key={region.id} value={region.id}>
+            {region.name}
+          </MenuItem>
+        ))}
+      </Select>
+      <Button className={sendButton} type="submit">
+        Send Message
+      </Button>
+    </form>
+  );
+};
+
+const AnnounceMessage = ({
+  toggleAnnounceMessage,
+  setToggleAnnounceMessage,
+}) => {
+  const iOS =
+    typeof navigator !== 'undefined' &&
+    /iPad|iPhone|iPod/.test(navigator.userAgent);
```

Comment:
Would it be useful to check if userAgent property exist before passing it as a test argument?